### PR TITLE
Eos 18876 UDX registration payload updates

### PR DESCRIPTION
**Pull request by Vrishali-web-dev**
# UI

 UI : Registration for Lyve Pilot - API payload updates and s3 URI addition.
 

## Problem Statement
<pre>
  <code>
    Story Ref (if any): EOS-18876 FE and BE : CORTX Lyve Pilot registration UI has not been updated with the most recent CSM/S3 service API changes
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes
Getting error on registration API which is under investigation from BE.
  </code>
</pre>
## Problem Description
<pre>
  <code>

The S3 API changes introduced to cortx-manager removed the URIs from the bucket creation and listing response payloads. Creating a new dedicated endpoint for these URIs, GET /api/v1/s3 .  
The Lyve Pilot registration workflow on cortx-management-portal is not up to date with these changes, and it requires there URIs to assemble the registration payload.  
Since they cannot be found on the original endpoints, the payload is incomplete and Lyve Pilot registration fails

  </code>
</pre>
## Solution/Screenshots
 <pre>
  <code>
- Added new GET /api/v1/s3 API to get S3 URI.
- Updated S3 URI to the device registration payload.
  </code>
</pre>
Response from /api/v1/s3 API:

image

S3 URI is updated in the registration Payload:

image

## Unit Test Cases
<pre>
  <code>
- Logged in to the S3 account and verified that no alerts API is getting called.
- User is now able to login with no any errors.

  </code>
</pre>

Signed-off-by: Vrishali Danave <vrishali.danave@seagate.com>

### CODE REVIEW

File: web/src/services/udx/routes.ts [hunk 1]
```diff
@@ -103,5 +103,20 @@ export default [
         }
       }
     ]
+  },
+  {
+    path: "/api/:version/s3",
+    method: "get",
+    handler: [
+      checkRequiredParams,
```
Comment (shrimetta) [inline web/src/services/udx/routes.ts:111]: pls also add checkApiVersion middleware

Comment (Vrishali-web-dev) [reply web/src/services/udx/routes.ts:112]: Added 'checkApiVersion' was missed in last commit

File: gui/src/components/udx/cortx-select-create-bucket.vue [hunk 2]
```diff
@@ -193,14 +196,11 @@ export default class CortxSelectCreateBucket extends Vue {
...
       this.bucketUrl = res && res.data.bucket_url ? res.data.bucket_url : "NA";
-      this.s3URI = res && res.data.bucket_url ? res.data.s3_uri : "NA";
 
       this.$store.dispatch("systemConfig/hideLoader");
     }
 
-    const URI: string = this.isCreateBucket
-      ? this.s3URI
-      : this.registrationForm.bucketName.s3URI;
+    const URI: string = this.s3URI;
```
Comment (shrimetta) [inline gui/src/components/udx/cortx-select-create-bucket.vue:203]: No need of URI variable. We can directly use this.s3URI in this.emit() like so (this.$emit("onChange", false, bucketName, this.s3URI);)

File: gui/src/components/udx/cortx-select-create-bucket.vue [hunk 3]
```diff
@@ -163,7 +163,10 @@ export default class CortxSelectCreateBucket extends Vue {
       this.bucketList = res.data.buckets.map((bucket: any) => {
         return { label: bucket.name, value: bucket.name, s3URI: bucket.s3_uri };
       });
-    }
+    };
+    const schemas = { schemas: "s3" }
```
Comment (shrimetta) [inline gui/src/components/udx/cortx-select-create-bucket.vue:167]: Missing semicolon. Please make sure there are no lint errors before making a commit.

**Review by shrimetta — CHANGES_REQUESTED** [review]
Please address minor comments

File: gui/src/components/udx/cortx-select-create-bucket.vue (hunk 3 above)
Comment (Vrishali-web-dev) [reply gui/src/components/udx/cortx-select-create-bucket.vue]: This is fixed

File: gui/src/components/udx/cortx-select-create-bucket.vue (hunk 2 above)
Comment (Vrishali-web-dev) [reply gui/src/components/udx/cortx-select-create-bucket.vue]: Updated the same.

**Review by JayshreeAmore — APPROVED** [review]

**Review by shrimetta — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**